spec/bolt_server/app_integration_spec.rb

```diff
@@ -99,5 +99,40 @@ def app
         expect(result['result']["stderr"]).to match(/not-a-command/)
       end
     end
+
+    describe "upload_file" do
```

Comment:
Any other test cases I should add here?

Reply:
If the directory already exists, this weird behavior happens: https://github.com/puppetlabs/bolt/issues/1120

Reply:
So does your implementation mirror this behaviour?

Reply:
Yes, because we want the implementation of this to mirror Bolt's behavior as closely as possible. When that issue gets fixed, it will be fixed here too. (It's inside `sudoable.rb`.)